### PR TITLE
PixelPaint: Let PickerTool optionally sample all layers

### PR DESCRIPTION
**Pull request by mustafaquraish**
Previously, only the color from the selected layer would be picked.
Now, we allow the user to select if they want to sample the color
from all layers.

### CODE REVIEW

File: Userland/Applications/PixelPaint/Image.cpp [hunk 1]
```diff
@@ -555,4 +556,14 @@ void Image::set_path(String path)
...
+Color Image::color_at(Gfx::IntPoint const& point) const
+{
+    Color color;
+    for (auto& layer : m_layers) {
+        if (layer.is_visible() && layer.rect().contains(point))
+            color = color.blend(layer.bitmap().get_pixel(point));
+    }
+    return color;
+}
+
```
Comment (metmo) [inline Userland/Applications/PixelPaint/Image.cpp:568]: Would it make sense to include opacity information for the layers as well? That way we get the actual shown color

Comment (mustafaquraish) [reply Userland/Applications/PixelPaint/Image.cpp:568]: Ah yes, you're right, I missed the layer opacity completely.

File: Userland/Applications/PixelPaint/Image.cpp [hunk 2]
```diff
@@ -560,8 +560,13 @@ Color Image::color_at(Gfx::IntPoint const& point) const
...
-        if (layer.is_visible() && layer.rect().contains(point))
-            color = color.blend(layer.bitmap().get_pixel(point));
+        if (!layer.is_visible() || !layer.rect().contains(point))
+            continue;
+
+        auto layer_color = layer.bitmap().get_pixel(point);
+        float layer_opacity = layer.opacity_percent() / 100.0f;
+        layer_color.set_alpha((u8)(layer_color.alpha() * layer_opacity));
+        color = color.blend(layer_color);
     }
```
Comment (metmo) [inline Userland/Applications/PixelPaint/Image.cpp:570]: Looks good to me and works as expected, but please squash this with the previous commit :)

Comment (mustafaquraish) [reply Userland/Applications/PixelPaint/Image.cpp:570]: Thanks for the feedback!

**Comment by awesomekling** [comment]
Great feature!

**Pull request merged**